prow/config/tide.go

```diff
@@ -83,6 +83,22 @@ type TidePriority struct {
 type Tide struct {
 	// SyncPeriod specifies how often Tide will sync jobs with GitHub. Defaults to 1m.
 	SyncPeriod *metav1.Duration `json:"sync_period,omitempty"`
+	// MaxGoroutines is the maximum number of goroutines spawned inside the
+	// controller to handle org/repo:branch pools. Defaults to 20. Needs to be a
+	// positive number.
+	MaxGoroutines int `json:"max_goroutines,omitempty"`
+	// BatchSizeLimitMap is a key/value pair of an org or org/repo as the key and
+	// integer batch size limit as the value. Use "*" as key to set a global default.
+	// Special values:
+	//  0 => unlimited batch size
+	// -1 => batch merging disabled :(
+	BatchSizeLimitMap map[string]int `json:"batch_size_limit,omitempty"`
+
+	TideGitHubConfig `json:",inline"`
```

Comment:
Inlining is nice for backwards compatibility and is probably the way to go for now, but I'm wondering if we'll have problems with naming collisions or confusion when we start adding Gerrit specific fields.

Reply:
We'll probably need to adjust if that ever happened, nice thing is that it won't affect the final interface between Prow and users